plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/hover/HoverProvider.java

```diff
@@ -76,31 +77,33 @@ public HoverProvider(EditorAgent editorAgent, TextDocumentServiceClient client,
         TextDocumentPositionParams paramsDTO = helper.createTDPP(document, context.getOffset());
 
         Promise<Hover> promise = client.hover(paramsDTO);
-        Promise<OrionHoverOverlay> then = promise.then(new Function<Hover, OrionHoverOverlay>() {
-            @Override
-            public OrionHoverOverlay apply(Hover arg) throws FunctionException {
-                OrionHoverOverlay hover = OrionHoverOverlay.create();
-                hover.setType("markdown");
-                String content = renderContent(arg);
-                // do not show hover with only white spaces
-                if (StringUtils.isNullOrWhitespace(content)) {
-                    return null;
-                }
-                hover.setContent(content);
-
-                return hover;
+        Promise<OrionHoverOverlay> then = promise.then((Function<Hover, OrionHoverOverlay>)arg -> {
+            OrionHoverOverlay hover = OrionHoverOverlay.create();
+            hover.setType("markdown");
+            String content = renderContent(arg);
+            // do not show hover with only white spaces
+            if (StringUtils.isNullOrWhitespace(content)) {
+                return null;
             }
+            hover.setContent(content);
 
-            private String renderContent(Hover hover) {
-                List<String> contents = new ArrayList<String>();
-                for (String dto : hover.getContents()) {
-                    // plain markdown text
-                    contents.add(dto);
-                }
-                return Joiner.on("\n\n").join(contents);
-            }
+            return hover;
         });
         return (JsPromise<OrionHoverOverlay>)then;
 
     }
+
+    private String renderContent(Hover hover) {
+        List<String> contents = new ArrayList<>();
+        for (Either<String, MarkedString> dto : hover.getContents()) {
+            if(dto.isLeft()) {
+                // plain markdown text
+                contents.add(dto.getLeft());
+            } else {
+                contents.add(dto.getRight().getLanguage());
```

Comment:
I think the "language" field is supposed to help render the content, not be displayed. Should we simply ignore the tag if we don't properly handle it?